config/kube.py

```diff
@@ -0,0 +1,392 @@
+        return
+
+
+    # Drop any active Cron job for join.
+    if drop_cron:
+        _drop_cron()
+
+    # Remove a key label and drain/delete self from cluster
+    # If not, the next join would fail
+    #
+    if os.path.exists(KUBE_ADMIN_CONF):
+        _label_node("enable_pods-")
+        run_command(
+                "kubectl --kubeconfig {} --request-timeout 20s drain {} --ignore-daemonsets".format(
+                    KUBE_ADMIN_CONF, hostname()),
+                ignore_error=True)
+        run_command(
+                "kubectl --kubeconfig {} --request-timeout 20s delete node {}".format(
+                    KUBE_ADMIN_CONF, hostname()),
+                ignore_error=True)
+
+    run_command("kubeadm reset -f", ignore_error=True)
+    run_command("rm -rf /etc/cni/net.d")
+    run_command("rm -f {}".format(KUBE_ADMIN_CONF))
+    run_command("systemctl stop kubelet")
+    run_command("systemctl disable kubelet")
+
+def kube_join(async_mode=False, force=False):
+    db_data = _get_db_data('KUBERNETES_MASTER', 'SERVER')
+    if not db_data or 'IP' not in db_data or not db_data['IP']:
+        _log_err("Kubernetes server is not configured")
+
+    if db_data['disable'].lower() != "false":
+        _log_err("kube join skipped as kubernetes server is marked disabled")
+        return
+
+    if not force:
+        if _is_connected(db_data['IP']):
+            # Already connected. No-Op
+            _drop_cron()
+            return
+
+    if async_mode:
+        _create_cron()
+    else:
+        kube_reset(drop_cron=False)
+        _do_join(db_data['IP'], db_data['insecure'])
+
+
+
+@click.group(cls=AbbreviationGroup)
+def kubernetes():
+    """kubernetes command line"""
+    pass
+
+# cmd kubernetes join [-a/--async] [-f/--force]
+@click.command()
+@click.option('-a', '--async', help='Join kubernetes cluster asynchronously', is_flag=True)
+@click.option('-f', '--force', help='Force a join', is_flag=True)
+def join(async, force):
+    kube_join(async_mode=async, force=force)
+kubernetes.add_command(join)
+
+# cmd kubernetes reset
+@click.command()
+def reset():
+    kube_reset()
+kubernetes.add_command(reset)
+
+
+# cmd kubernetes server
+@click.group()
+def server():
+    """ Server configuration """
+    pass
+kubernetes.add_command(server)
+
+
+#cmd kubernetes server IP
+@click.command()
+@click.argument('vip')
+def ip(vip):
+    """Specify a kubernetes ckuster VIP"""
+    if not netaddr.IPAddress(vip):
+        click.echo('Invalid IP address %s' % vip)
+        return
+    _update_kube_server('IP', vip)
+server.add_command(ip)
+
+#cmd kubernetes server insecure
+@click.command()
+@click.argument('option', type=click.Choice(["True", "False"]))
+def insecure(option):
+    """Specify a kubernetes ckuster VIP access as insecure or not"""
+    _update_kube_server('insecure', option)
+server.add_command(insecure)
+
+#cmd kubernetes server disable
+@click.command()
+@click.argument('option', type=click.Choice(["True", "False"]))
```

Comment:
We don't use "True"/"False" choices in any other commands that I'm aware of. Suggest changing to "on"/"off" to align with other commands?

Reply:
Switched to On/Off